node/silkworm/stagedsync/stage_execution.cpp

```diff
@@ -278,31 +276,28 @@ StageResult unwind_execution(db::RWTxn& txn, const std::filesystem::path&, uint6
     };
 
     try {
-        if (unwind_to == 0) {
-            for (const auto& unwind_table : unwind_tables) {
-                auto unwind_map_handle{db::open_map(*txn, unwind_table)};
-                txn->clear_map(unwind_map_handle);
-            }
-        } else {
-            {
-                auto plain_state_table{db::open_cursor(*txn, db::table::kPlainState)};
-                auto plain_code_table{db::open_cursor(*txn, db::table::kPlainContractCode)};
-                auto account_changeset_table{db::open_cursor(*txn, db::table::kAccountChangeSet)};
-                auto storage_changeset_table{db::open_cursor(*txn, db::table::kStorageChangeSet)};
-                unwind_state_from_changeset(account_changeset_table, plain_state_table, plain_code_table, unwind_to);
-                unwind_state_from_changeset(storage_changeset_table, plain_state_table, plain_code_table, unwind_to);
-            }
 
-            // Delete records which has keys greater than unwind point
-            // Note erasing forward the start key is included that's why we increase unwind_to by 1
-            Bytes start_key(8, '\0');
-            endian::store_big_u64(&start_key[0], unwind_to + 1);
-            for (int i = 2; i < 7; ++i) {
-                auto unwind_cursor{db::open_cursor(*txn, unwind_tables[i])};
-                auto erased{db::cursor_erase(unwind_cursor, start_key, db::CursorMoveDirection::Forward)};
-                log::Info() << "Erased " << erased << " records from " << unwind_tables[i].name;
-                unwind_cursor.close();
+        {
+            // Revert states
+            auto plain_state_table{db::open_cursor(*txn, db::table::kPlainState)};
+            auto plain_code_table{db::open_cursor(*txn, db::table::kPlainContractCode)};
+            auto account_changeset_table{db::open_cursor(*txn, db::table::kAccountChangeSet)};
+            auto storage_changeset_table{db::open_cursor(*txn, db::table::kStorageChangeSet)};
+            unwind_state_from_changeset(account_changeset_table, plain_state_table, plain_code_table, unwind_to);
+            unwind_state_from_changeset(storage_changeset_table, plain_state_table, plain_code_table, unwind_to);
+        }
+
+        // Delete records which has keys greater than unwind point
+        // Note erasing forward the start key is included that's why we increase unwind_to by 1
+        Bytes start_key(8, '\0');
+        endian::store_big_u64(&start_key[0], unwind_to + 1);
+        for (const auto& map_config : unwind_tables) {
+            auto unwind_cursor{db::open_cursor(*txn, map_config)};
+            auto erased{db::cursor_erase(unwind_cursor, start_key, db::CursorMoveDirection::Forward)};
+            if(erased > 16) {
+                log::Info() << "Erased " << erased << " records from " << map_config.name;
             }
+            unwind_cursor.close();
```

Comment:
Will be closed automatically by the destructor, won't it?

Reply:
Yeah right